octodns/provider/ns1.py

```diff
@@ -298,6 +309,34 @@ def __init__(self, id, api_key, retry_count=4, monitor_regions=None,
         self._client = Ns1Client(api_key, parallelism, retry_count,
                                  client_config)
 
+    # Allowed filter configurations:
+    # 1. Filter chain is the basic filter chain
+    # 2. In addition, it could have the 'geofence_country' filter and or
+    # 'geofence_regional' filter only at index 1 in the filter chain
+    # 3. If both country and regional filters are present, they are required
+    # to be as below:
+    #   - 'geofence_regional' at index 1
+    #   - 'geofence_country' at index 2
+    # Any other deviation is returned as an unsupported filter configuration
+    def _valid_filter_config(self, filter_cfg):
+        has_region = 'geofence_regional' in [f['filter'] for f in filter_cfg]
+        has_country = 'geofence_country' in [f['filter'] for f in filter_cfg]
+        return filter_cfg == self._get_updated_filter_chain(has_region,
+                                                            has_country)
+
+    def _get_updated_filter_chain(self, has_region, has_country):
+        filters = deepcopy(self._BASIC_DYNAMIC_FILTERS)
```

Comment:
Not sure a `deepcopy` is necssary here since you're not modifynig the items themselves, just (maybe) adding/removing items from the list. You can probably just do:

```suggestion
        filters = list(self._BASIC_DYNAMIC_FILTERS)
```

Reply:
Fixed.